system-contracts/contracts/DefaultAccount.sol

```diff
@@ -139,6 +139,12 @@ contract DefaultAccount is IAccount {
         bytes calldata data = _transaction.data;
         uint32 gas = Utils.safeCastToU32(gasleft());
 
+        // TODO: if possible, maybe implement some way to avoid memory copying here.
+        if (_transaction.reserved[1] != 0) {
```

Comment:
Shouldn't we also add `to == address(0)` here?

Reply:
Yes, I believe in practice it should be redundant since the operator only sets the reserved `1` flag to one in case of transactions to the zero address but it's a good idea to add it as a safety check. Added it in [0ecbe61](https://github.com/matter-labs/era-contracts/pull/685/commits/0ecbe61e0794ae9e4c82ff230e13e14ea91d84a0)